src/meta/src/hummock/manager/mod.rs

```diff
@@ -2459,6 +2404,235 @@ where
+                            }) => {
+                                assert_ne!(0, pull_task_count);
+                                if let Some(compactor) = hummock_manager.compactor_manager.get_compactor(context_id) {
+                                    use rand::prelude::SliceRandom;
+                                    use rand::thread_rng;
+                                    let mut compaction_group_ids = hummock_manager.compaction_group_ids().await;
+                                    compaction_group_ids.shuffle(&mut thread_rng());
+
+                                    let (group, task_type) = {
+                                        let (mut group,mut task_type) = (None, None);
+                                        for cg_id in compaction_group_ids {
+                                            if let Some(pick_type) = hummock_manager.compaction_state.auto_pick_type(cg_id) {
+                                                group = Some(cg_id);
+                                                task_type = Some(pick_type);
+
+                                                break;
+                                            }
+                                        }
+
+                                        (group, task_type)
+                                    };
+
+                                    if let (Some(group), Some(task_type)) = (group, task_type) {
+                                        let selector: &mut Box<dyn LevelSelector> = compaction_selectors.get_mut(&task_type).unwrap();
+                                        for _ in 0..pull_task_count {
+                                            let compact_task =
+                                                hummock_manager
+                                                .get_compact_task(group, selector)
+                                                .await;
+
+                                            match compact_task {
+                                                Ok(Some(compact_task)) => {
+                                                    let task_id = compact_task.task_id;
+                                                    if let Err(e) = compactor.send_event(
+                                                        ResponseEvent::CompactTask(compact_task)
+                                                    ) {
+                                                        tracing::warn!(
+                                                            "Failed to send task {} to {}. {:#?}",
+                                                            task_id,
+                                                            compactor.context_id(),
+                                                            e
+                                                        );
+
+                                                        // try cancel on compactor
+                                                        let _ = compactor.cancel_task(task_id);
```

Comment:
I think we should at least log the error instead of ignoreing it.

Reply:
We've already printed a log for send_task failure, so I'm assuming that the cancel_task here is likely to fail, so can we just remove the cancel_task here (since the connection has failed)?